hole/poker.go

```diff
@@ -170,11 +170,40 @@ func poker() (args []string, out string) {
 		hands = append(hands, Hand{"Royal Flush", hand})
 	}
 
+	// For a flush, the highest minus lowest codepoint is at most 13, but this is not sufficient
+	// for detecting a flush. Generate hands that meet this criteria that aren't flushes.
+	for suit := 0; suit < 3; suit++ {
+		// Start near the top of the range on one of the lower three suits.
+		// The hand will have cards with two different suits and five different face values.
+		// The start card is at least 10 to avoid a straight.
+		startCard := 12 - rand.Intn(3)
+		// The offset of the end card from the start card must be at least 4.
+		// It should be at most 11 to ensure that the range in codepoints doesn't exceed 13.
+		endOffset := 4 + rand.Intn(7)
+		offsets := rand.Perm(endOffset)
+		hand := []rune{cardRune(startCard, suit)}
+		for _, offset := range offsets[:4] {
+			card := startCard + offset + 1
+			hand = append(hand, cardRune(card%13, suit+card/13))
+		}
+		hands = append(hands, Hand{"High Card", hand})
+	}
+
+	// High Card, but could be mistaken for a straight.
+	for suit := 0; suit < 3; suit++ {
+		hands = append(hands, Hand{"High Card", []rune{
+			cardRune(12, suit),
+			cardRune(0, suit+1),
+			cardRune(1, suit+1),
+			cardRune(2, suit+1),
+			cardRune(3, suit+1),
+		}})
+	}
+
 	// Shuffle the hands.
-	for i := range hands {
-		j := rand.Intn(i + 1)
+	rand.Shuffle(len(hands), func(i, j int) {
```

Comment:
That's a nice idiom, that can probably be used all over the shop.

Reply:
Looks like its faster too, I must've missed this in the 1.10 changelog https://stackoverflow.com/a/46185753

Reply:
It looks like it's essentially identical but with one fewer loop iteration.
The problem is the linter doesn't like capturing the hand range variable and I don't think we can make linter exceptions.

Reply:
Interesting. I presume this is standard `go lint` rather than the golintci that's ran by github action and `make lint`?

Reply:
This is the one github runs. Maybe we can make an exception, I'm not sure.

Reply:
Oh you definitely can, and I've done it before, there's a reason these checks aren't part of Go proper. Just edit `.golangci.toml`, you can run the linter locally with `make lint`

Reply:
I disabled scopelint.